src/client/pages/components/Login.js

```diff
@@ -1,41 +1,153 @@
-import React from 'react';
+import React, { Component, useState } from 'react';
 
+import { Link } from 'react-router-dom';
 import './style/Login.css';
+import ReactModal from 'react-modal';
+
+function Login() {
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [username, setUsername] = useState('');
+  const [showLoginModal, setShowLoginModal] = useState(false);
+  const [showRegistrationModal, setShowRegistrationModal] = useState(false);
+
+  function handleSubmit(event) {
 
-class Login extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      loggedIn: false,
-      username: 'Username',
-    }
-    this.handleClick = this.handleClick.bind(this);
   }
 
-  handleClick() {
+  function handleOpenLoginModal() {
+    handleCloseRegistrationModal();
+    if (!showLoginModal) setShowLoginModal(true);
+    // document.body.style.overflowY = 'hidden';
+  }
 
+  function handleCloseLoginModal() {
+    setShowLoginModal(false);
+    document.body.style.overflowY = 'unset';
   }
 
-  render() {
-    if (!this.state.loggedIn) {
-      return (
-        <div>
-          <nav className="login">
-            <a href="/">Login</a>
-            <a className="profile-signup" href="" onClick={this.handleClick}>Sign Up</a>
-          </nav>
-        </div>
-      );
-    } else {
-      return (
-        <div>
-          <nav className="profile">
-            <a href="/">{this.state.username}</a>
-          </nav>
+  function handleOpenRegistrationModal() {
+    handleCloseLoginModal();
+    if (!showRegistrationModal) setShowRegistrationModal(true);
+    // document.body.style.overflowY = 'hidden';
+  }
+
+  function handleCloseRegistrationModal() {
+    setShowRegistrationModal(false);
+    document.body.style.overflowY = 'unset';
+  }
+
+  function handleRegistrationClick() {
+    setShowLoginModal(false);
+    setShowRegistrationModal(true);
+  }
+
+
+  function handleClick() {
+
+  }
+
+  if (!loggedIn) {
+    return (
```

Comment:
Could you make the modal component into its own rather than returning in the if  statement?